src/NzbDrone.Core/IndexerSearch/EpisodeSearchService.cs

```diff
@@ -79,36 +115,34 @@ public void Execute(EpisodeSearchCommand message)
 
         public void Execute(MissingEpisodeSearchCommand message)
         {
-            //TODO: Look at ways to make this more efficient (grouping by series/season)
+            List<Episode> episodes;
 
-            var episodes =
-                _episodeService.EpisodesWithoutFiles(new PagingSpec<Episode>
-                                                     {
-                                                         Page = 1,
-                                                         PageSize = 100000,
-                                                         SortDirection = SortDirection.Ascending,
-                                                         SortKey = "Id",
-                                                         FilterExpression = v => v.Monitored == true && v.Series.Monitored == true
-                                                     }).Records.ToList();
-
-            var missing = episodes.Where(e => !_queueService.GetQueue().Select(q => q.Episode.Id).Contains(e.Id)).ToList();
-
-            _logger.ProgressInfo("Performing missing search for {0} episodes", missing.Count);
-            var downloadedCount = 0;
+            if (message.SeriesId > 0)
+            {
+                episodes = _episodeService.GetEpisodeBySeries(message.SeriesId)
+                                          .Where(e => e.Monitored && !e.HasFile)
+                                          .ToList();
+            }
 
-            //Limit requests to indexers at 100 per minute
-            using (var rateGate = new RateGate(100, TimeSpan.FromSeconds(60)))
+            else
             {
-                foreach (var episode in missing)
-                {
-                    rateGate.WaitToProceed();
-                    var decisions = _nzbSearchService.EpisodeSearch(episode);
-                    var processed = _processDownloadDecisions.ProcessDecisions(decisions);
-                    downloadedCount += processed.Grabbed.Count;
-                }
+                episodes = _episodeService.EpisodesWithoutFiles(new PagingSpec<Episode>
+                                                                    {
+                                                                        Page = 1,
+                                                                        PageSize = 100000,
+                                                                        SortDirection = SortDirection.Ascending,
+                                                                        SortKey = "Id",
+                                                                        FilterExpression =
+                                                                            v =>
+                                                                            v.Monitored == true &&
+                                                                            v.Series.Monitored == true
```

Comment:
does this work? isn't this the same filter used for the missing page?

is this method ever used with unmonitored/series episdoes? if not we should make the filter built in and only pass in the other paging specs.